In [1]:
import graphlab

In [2]:
song_data = graphlab.SFrame('song_data.gl/')

[INFO] graphlab.cython.cy_server: GraphLab Create v2.1 started. Logging: /tmp/graphlab_server_1536904626.log


This non-commercial license of GraphLab Create for academic use is assigned to r38411@yahoo.com and will expire on August 31, 2019.


In [3]:
song_data.head()

Columns:
	user_id	str
	song_id	str
	listen_count	int
	title	str
	artist	str
	song	str

Rows: 10

Data:
+-------------------------------+--------------------+--------------+
|            user_id            |      song_id       | listen_count |
+-------------------------------+--------------------+--------------+
| b80344d063b5ccb3212f76538f... | SOAKIMP12A8C130995 |      1       |
| b80344d063b5ccb3212f76538f... | SOBBMDR12A8C13253B |      2       |
| b80344d063b5ccb3212f76538f... | SOBXHDL12A81C204C0 |      1       |
| b80344d063b5ccb3212f76538f... | SOBYHAJ12A6701BF1D |      1       |
| b80344d063b5ccb3212f76538f... | SODACBL12A8C13C273 |      1       |
| b80344d063b5ccb3212f76538f... | SODDNQT12A6D4F5F7E |      5       |
| b80344d063b5ccb3212f76538f... | SODXRTY12AB0180F3B |      1       |
| b80344d063b5ccb3212f76538f... | SOFGUAY12AB017B0A8 |      1       |
| b80344d063b5ccb3212f76538f... | SOFRQTD12A81C233C0 |      1       |
| b80344d063b5ccb3212f76538f... | SOHQWYZ12A6D4FA701 |      1       |
+-------------------------------+--------------------+--------------+
+-------------------------------+-------------------------------+
|             title             |             artist            |
+-------------------------------+-------------------------------+
|            The Cove           |          Jack Johnson         |
|        Entre Dos Aguas        |         Paco De Lucia         |
|            Stronger           |           Kanye West          |
|         Constellations        |          Jack Johnson         |
|          Learn To Fly         |          Foo Fighters         |
|  Apuesta Por El Rock 'N' Roll |      Héroes del Silencio      |
|         Paper Gangsta         |           Lady GaGa           |
|         Stacked Actors        |          Foo Fighters         |
|         Sehr kosmisch         |            Harmonia           |
| Heaven's gonna burn your eyes | Thievery Corporation feat.... |
+-------------------------------+-------------------------------+
+-------------------------------+
|              song             |
+-------------------------------+
|    The Cove - Jack Johnson    |
| Entre Dos Aguas - Paco De ... |
|     Stronger - Kanye West     |
| Constellations - Jack Johnson |
|  Learn To Fly - Foo Fighters  |
| Apuesta Por El Rock 'N' Ro... |
|   Paper Gangsta - Lady GaGa   |
| Stacked Actors - Foo Fighters |
|    Sehr kosmisch - Harmonia   |
| Heaven's gonna burn your e... |
+-------------------------------+
[10 rows x 6 columns]

# 1. use unique to count unique users for each artist: Kanye West, Foo Fighers, Taylor Swift, Lady Gaga

In [8]:
artist_list = ['Kanye West', 'Foo Fighters', 'Taylor Swift', 'Lady GaGa']
user_artist = {}
for i in artist_list:
    user_artist[i] = len((song_data[song_data['artist']==i]).unique())
user_artist

{'Foo Fighters': 3429,
 'Kanye West': 3775,
 'Lady GaGa': 4129,
 'Taylor Swift': 6227}

## Unique user for the artist: Foo Fighters: 3429, Kanye: 3775, Lady: 4129, Taylor: 6227

# 2. use group_by to find listen count

In [12]:
art_list_count = song_data.groupby(key_columns='artist', operations={'total_count':graphlab
                                                    .aggregate.SUM('listen_count')})
art_list_count.sort('total_count', ascending=False)

Columns:
	artist	str
	total_count	int

Rows: 3375

Data:
+------------------------+-------------+
|         artist         | total_count |
+------------------------+-------------+
|     Kings Of Leon      |    43218    |
|     Dwight Yoakam      |    40619    |
|         Björk          |    38889    |
|        Coldplay        |    35362    |
| Florence + The Machine |    33387    |
|     Justin Bieber      |    29715    |
|    Alliance Ethnik     |    26689    |
|      OneRepublic       |    25754    |
|         Train          |    25402    |
|     The Black Keys     |    22184    |
+------------------------+-------------+
[3375 rows x 2 columns]
Note: Only the head of the SFrame is printed.
You can use print_rows(num_rows=m, num_columns=n) to print more rows and columns.